### DISCUSSION TITLE
Be able to specify or inherit rendering device

### DISCUSSION THREAD

**Discussion by Bestest-Coder** [godotengine/godot-proposals#6504] [Editor]
I develop on a Linux laptop and would very much like to have an easy way to specify from the project chooser dialog or otherwise have the debug runs of projects or scenes in the editor inherit if a certain rendering device (i.e. integrated vs discrete graphics) are used when specified with something like the `--gpu-index 0` CLI option. Currently it always defaults to my discrete graphics card, I do not find specifying `--gpu-index` to be a bad design choice, but when something like that is explicitly requested by the user it should be inherited by further project debug instances.

**Comment by AThousandShips** [comment]
You can do this in the `editor/run/main_run_args` project setting (probably under advanced settings), and there could be issues with applying the same options to the running instance as the editor one and as it's easy to supply the specified settings it might be safer that way

**Reply by AThousandShips** [reply]
Also some options are forwarded, but being able to specify rendering device would be a useful setting to add

**Reply by AThousandShips** [reply]
My bad I missed that this wasn't forwarded by the project manager to the editor

**Comment by Calinou** [comment]
Feel free to open a pull request for this :slightly_smiling_face: 

Forwarding CLI arguments is done from the project manager to the editor manually, which means it involves a lot of copy-pasted code that accumulated over time unfortunately. This is why it's not uniformly done for all CLI arguments.

**Comment by Bestest-Coder** [comment]
Pull request has been opened at [#75198](https://github.com/godotengine/godot/pull/75198)
